### PR TITLE
Balance search pages to have view mode-aware max search results

### LINKED ISSUE #829: Changing page size doesn't change page number
### Describe the bug

(In mod browsing)

### Steps to reproduce

1. Go to page, let's say, 4,578.
2. Change the page size from 20 items per page to 100
3. The site remains on page 4,578, and it doesn't show any projects, as all pages are five times larger now

### Expected behavior

When changing page size, the current page should change to `newpage = currentpage / (newpagesize / currentpagesize)`, for example `4,578 / (100 / 20)`.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Prospector**
On grid mode, it will have these options which are even multiples of 3:
image

While gallery mode has all even options:
image

Also fixes #829

### CODE REVIEW

File: pages/search.vue [hunk 1]
```diff
@@ -841,6 +853,21 @@ export default {
...
         $cookies: this.$cookies,
       })
+      this.setClosestMaxResults()
+    },
+    setClosestMaxResults() {
+      const view = this.$cosmetics.searchDisplayMode[this.projectType.id]
+      const maxResultsOptions = this.maxResultsForView[view] ?? [20]
+      const currentMax = this.maxResults
+      console.log(maxResultsOptions)
+      console.log(currentMax)
```
Comment (Geometrically) [inline pages/search.vue:863]: Should be removed

**Review by Geometrically — CHANGES_REQUESTED** [review]
Looks good besides console statements

**Review by Geometrically — CHANGES_REQUESTED** [review]
One more console statement but good to merge :)

File: pages/search.vue [hunk 2]
```diff
@@ -679,6 +692,20 @@ export default {
...
+      newPageNumber = Math.max(
+        1,
+        Math.min(
+          Math.floor(
+            newPageNumber / (this.maxResults / this.previousMaxResults)
+          ),
+          this.pageCount
+        )
+      )
+      console.log(newPageNumber)
```
[the remark this answers is not in the document]
Comment (Prospector) [reply pages/search.vue]: dammit, fixed.

**Pull request merged**